### PR TITLE
Rc interpolation optimization

### PR DESCRIPTION
**Pull request by orneryd**
this change reduces CPU load considerably when using rc_interpolation.

### CODE REVIEW

File: src/main/fc/fc_rc.c [hunk 1]
```diff
@@ -179,42 +188,19 @@ static void checkForThrottleErrorResetState(uint16_t rxRefreshRate)
 
 void processRcCommand(void)
 {
-    static float rcCommandInterp[4] = { 0, 0, 0, 0 };
```
Comment (kidBrazil) [inline src/main/fc/fc_rc.c]: Cant believe all of this was being set each time the loop ran.

**Comment by orneryd** [comment]
verified in conjunction with the imuf_104 branch. This reduces CPU load considerably for 32K. It limits the interpolation to every other loop when running faster than 16K.

**Comment by rs2k** [comment]
Woohoo! Say goodbye to redundant division and hello to speed! 🥇 

Looks good to me!

File: src/main/fc/fc_rc.c [hunk 2]
```diff
@@ -179,49 +190,43 @@ static void checkForThrottleErrorResetState(uint16_t rxRefreshRate)
...
+                    rxRefreshRate = 1000 * rxConfig()->rcInterpolationInterval;
+                    break;
+                case RC_SMOOTHING_OFF:
+                case RC_SMOOTHING_DEFAULT:
+                default:
+                    rxRefreshRate = rxGetRefreshRate();
+            }
 
-        if (isRXDataNew && rxRefreshRate > 0) {
             rcInterpolationStepCount = rxRefreshRate / targetPidLooptime;
```
Comment (rs2k) [inline src/main/fc/fc_rc.c]: targetPidLooptime should have an inverse recalculated somewhere.

Comment (orneryd) [reply src/main/fc/fc_rc.c]: as far as I can tell it's a float that is 31.25, 62.5, 125 etc..

File: src/main/fc/fc_rc.c (hunk 1 above)
Comment (wind0r) [reply src/main/fc/fc_rc.c]: kidBrazil it looks like you didn't notice the static. Something like this is in plenty of functions and other firmwares. The set doesn't happen with every loop

**Pull request merged**